### PR TITLE
Add support Terraform Cloud / Terraform Enterprise

### PR DESCRIPTION
**Pull request by cohalz**
This is a [Terraform Cloud / Terraform Enterprise backend](https://www.terraform.io/docs/language/settings/backends/remote.html). Now it is possible to use terraform.tfstate file with `"type": "remote"` backend. Additionaly it supports `remote://` schema. (e.g. `"remote://app.terraform.io/myorg/myworkspace"`)

This PR should solve #29

### CODE REVIEW

File: tfstate/remote_tfe.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	var err error
+	var client *tfe.Client
+	if token != "" {
+		client, err = tfe.NewClient(&tfe.Config{
+			Address: address,
+			Token:   token,
+		})
+	} else {
+		client, err = tfe.NewClient(&tfe.Config{
+			Address: address,
```
Comment (cohalz) [inline tfstate/remote_tfe.go:48]: In this case, use the environment variable `TFE_TOKEN`.
https://registry.terraform.io/providers/hashicorp/tfe/latest/docs

File: tfstate/remote_tfe.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	}
+
+	return nil, errors.Errorf("workspaces requires either name or prefix")
+}
+
+func readTFE(hostname string, organization string, ws string, token string) (io.ReadCloser, error) {
+	var address string
+	address = tfe.DefaultAddress
+	if hostname != "" {
+		address = "https://" + hostname
```
Comment (cohalz) [inline tfstate/remote_tfe.go:36]: Terraform Enterprise requires a TLS connection.

> The Terraform CLI performs a TLS handshake with Terraform Enterprise during service discovery, and will need access to the Certificate Authority that Terraform Enterprise uses.
https://www.terraform.io/docs/enterprise/install/installer.html

**Comment by fujiwara** [comment]
@cohalz Great! I confirmed this PR with my workspace on Terraform Cloud. Thank you!

File: tfstate/remote_tfe.go (hunk 1 above)
Comment (emilhdiaz) [reply tfstate/remote_tfe.go:48]: The TFE provider is different than using Terraform Cloud / Enterprise as a backend. The `TFE_TOKEN` environment variable still works on the provider, but not the backend configuration. 

See for more info: https://github.com/hashicorp/terraform/issues/21965

**Pull request merged**